pool/src/contract.rs

```diff
@@ -271,6 +280,17 @@ impl Pool for PoolContract {
             .publish((Symbol::new(&e, "update_pool"), admin), backstop_take_rate);
     }
 
+    fn set_max_positions(e: Env, max: u32) {
+        storage::extend_instance(&e);
+        let admin = storage::get_admin(&e);
+        admin.require_auth();
+
+        storage::set_max_positions(&e, &max);
+
+        e.events()
+            .publish((Symbol::new(&e, "set_max_positions"), admin), max);
+    }
```

Comment:
this should likely be included with `pool_init_meta`, right? Like backstopTakeRate, it feels like a vital piece of setup required for the pool to function and should be set on initialization.

In the same vein, it feels appropriate to include this with `update_pool` or rename that function to `set_backstop_rate` to avoid confusion.

Having one function that sets multiple pieces of pool configuration doesn't seem problematic to me, but I don't have a strong opinion either way.